### PR TITLE
Keep track of unsaved changes in the application and pass that to the host application

### PR DESCRIPTION
**Pull request by Dananji**
To keep track of unsaved changes in a host application a function is passed into SME as a prop, which will pass the status of the structure to the hosting application. This can be then used to show the confirmation when closing the modal.

@adamjarling This PR is huge, because I changed a state variable name and value (`smData` -> `structuralMetadata` including 2 values `smData` and `initSmData`) in Redux store which I then had to change in all the components. This `initSmData` is added to keep track of the initial structure, so that we can compare edited structure with the initial one to identify unsaved changes.

### CODE REVIEW

File: src/actions/sm-data.js [hunk 1]
```diff
@@ -1,4 +1,12 @@
 import * as types from './types';
+import { updateStructureStatus } from './forms';
+
+export function reBuildSMUI(items) {
+  return dispatch => {
+    dispatch(buildSMUI(items));
+    dispatch(updateStructureStatus(0));
```
Comment (adamjarling) [inline src/actions/sm-data.js]: Is there a comment block somewhere which details what `code` values mean for this function?    For example, if I were looking at this for the first time, I'd assume `updateStructureStatus(0)` is sending a `false` argument, `0`.

Comment (Dananji) [reply src/actions/sm-data.js]: No it does not have any details. Yeah documenting that is a good idea :+1: I will add a comment in actions explaining the code.

**Comment by Dananji** [comment]
@adamjarling The logic for identifying unsaved changes goes like this;

1. No changes made -> modal closes without confirmation
2. Changes made and not saved -> confirmation is required
3. Change made and revert it (nothing changed at the end) -> modal closes without confirmation
4. Changes made and saved -> modal closes without confirmation

Since we don't have any undo/redo functionality for SME, I'm not sure about the 3rd one? What do you think?

**Comment by adamjarling** [comment]
@Dananji Hmmm, maybe we could chat through this for 10 min this afternoon?

**Pull request merged**